packages/@aws-cdk/cloudformation-include/lib/cfn-include.ts

```diff
@@ -0,0 +1,128 @@
+ * You can retrieve any of the resources defined in the template file by calling the {@link getResource} method.
+ * Any modifications you make on the returned resource objects will be reflected in the resulting CDK template.
+ *
+ * @see getResource
+ */
+export class CfnInclude extends core.CfnElement {
+  private readonly resources: { [logicalId: string]: core.CfnResource } = {};
+  private readonly template: any;
+  private readonly preserveLogicalIds: boolean;
+
+  constructor(scope: core.Construct, id: string, props: CfnIncludeProps) {
+    super(scope, id);
+
+    // read the template into a JS object
+    this.template = futils.readJsonSync(props.templateFile);
+
+    // ToDo implement preserveLogicalIds=false
+    this.preserveLogicalIds = true;
+
+    // instantiate all resources as CDK L1 objects
+    for (const logicalId of Object.keys(this.template.Resources || {})) {
+      this.getOrCreateResource(logicalId);
+    }
+  }
+
+  /**
+   * Returns the low-level CfnResource from the template with the given logical ID.
+   * Any modifications performed on that resource will be reflected in the resulting CDK template.
+   *
+   * The returned object will be of the proper underlying class;
+   * you can always cast it to the correct type in your code:
+   *
+   *     // assume the template contains an AWS::S3::Bucket with logical ID 'Bucket'
+   *     const cfnBucket = cfnTemplate.getResource('Bucket') as s3.CfnBucket;
+   *     // cfnBucket is of type s3.CfnBucket
+   *
+   * If the template does not contain a resource with the given logical ID,
+   * an exception will be thrown.
+   *
+   * @param logicalId the logical ID of the resource in the CloudFormation template file
+   */
+  public getResource(logicalId: string): core.CfnResource {
+    const ret = this.resources[logicalId];
+    if (!ret) {
+      throw new Error(`Resource with logical ID '${logicalId}' was not found in the template`);
+    }
+    return ret;
+  }
+
+  /** @internal */
+  public _toCloudFormation(): object {
+    const ret: { [section: string]: any } = {};
+
+    for (const section of Object.keys(this.template)) {
+      // render all sections of the template unchanged,
+      // except Resources, which will be taken care of by the created L1s
+      if (section !== 'Resources') {
+        ret[section] = this.template[section];
+      }
+    }
+
+    return ret;
+  }
+
+  private getOrCreateResource(logicalId: string): core.CfnResource {
+    const ret = this.resources[logicalId];
+    if (ret) {
+      return ret;
+    }
+
+    const resourceAttributes: any = this.template.Resources[logicalId];
+    const l1ClassFqn = cfn_type_to_l1_mapping.lookup(resourceAttributes.Type);
+    if (l1ClassFqn) {
+      // fail early for resource attributes we don't support yet
+      const unsupportedAttributes = ['Condition', 'DependsOn', 'CreationPolicy', 'UpdatePolicy'];
```

Comment:
Might be better to do an whitelist rather than a blacklist, so we don't miss new attributes that CF may add, or existing (rare) ones that we're not aware of.

Reply:
Sure. Done.